packages/integration-karma/test/mixed-shadow-mode/transitivity/index.spec.js

```diff
@@ -0,0 +1,117 @@
+            elm = createElement('x-native-light-synthetic', { is: NativeLightSynthetic });
+            document.body.appendChild(elm);
+        });
+
+        afterEach(() => {
+            setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', false);
+            setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+        });
+
+        it('should attach shadow root (native component)', () => {
+            if (process.env.NATIVE_SHADOW_ROOT_DEFINED) {
+                expect(isNativeShadowRootInstance(elm.shadowRoot)).toBeTrue();
+            } else {
+                expect(isSyntheticShadowRootInstance(elm.shadowRoot)).toBeTrue();
+            }
+        });
+
+        it('should not attach shadow root (light composed within native)', () => {
+            const light = elm.shadowRoot.querySelector('x-light-synthetic');
+            expect(light.shadowRoot).toBeNull();
+        });
+
+        it('should attach shadow root (synthetic composed within light composed within native)', () => {
+            const light = elm.shadowRoot.querySelector('x-light-synthetic');
+            const synthetic = light.querySelector('x-synthetic');
+            if (process.env.NATIVE_SHADOW_ROOT_DEFINED) {
+                expect(isNativeShadowRootInstance(synthetic.shadowRoot)).toBeTrue();
+            } else {
+                expect(isSyntheticShadowRootInstance(synthetic.shadowRoot)).toBeTrue();
+            }
+        });
+
+        it('should attach shadow root (synthetic assigned to light slot composed within native)', () => {
+            const light = elm.shadowRoot.querySelector('x-light-synthetic');
+            const synthetic = light.querySelector('x-synthetic.slot-assigned');
+            if (process.env.NATIVE_SHADOW_ROOT_DEFINED) {
+                expect(isNativeShadowRootInstance(synthetic.shadowRoot)).toBeTrue();
+            } else {
+                expect(isSyntheticShadowRootInstance(synthetic.shadowRoot)).toBeTrue();
+            }
+        });
+    });
+
+    describe('when root component renderMode="light"', () => {
+        let elm;
+
+        beforeEach(() => {
+            setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', true);
+            setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', true);
+            elm = createElement('x-light-synthetic', { is: LightSynthetic });
+            document.body.appendChild(elm);
+        });
+
+        afterEach(() => {
+            setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', false);
+            setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+        });
+
+        it('should not attach any shadow root (root component)', () => {
+            expect(elm.shadowRoot).toBeNull();
+        });
+
+        it('should attach shadow root (syntheic composed within light)', () => {
+            const synthetic = elm.querySelector('x-synthetic');
+            expect(isSyntheticShadowRootInstance(synthetic.shadowRoot)).toBeTrue();
+        });
```

Comment:
I wonder if it would make sense to add this case?

```html
<light>
  <shadow-any>
    #shadow-root
      <slot>
        <shadow-default></shadow-default>
      </slot>
  </shadow-any>
</light>
```

In this case, one might naïvely think that the `<shadow-default>` component should be native, but it should actually be synthetic, since it was slotted into the `<shadow-any>` component by the `<light>` component.

Reply:
Added!